### PR TITLE
add store

### PR DESCRIPTION
**Pull request by LucaMele**

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      if (typeof value === 'object') {
+        this.store[key] = this.deepClone({
+          ...this.store[key],
+          ...value,
+        });
+      } else {
+        this.store[key] = value;
+      }
+    } else {
+      this.store = this.deepClone({
```
Comment (AndyOGo) [inline src/store/index.ts]: Why deep cloning the whole store here?
Can't it be done like above?

```js
 this.store[key] = this.deepClone(value);
```

Comment (LucaMele) [reply src/store/index.ts]: yepp agree. Thanks :)

File: src/store/index.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    this.store[key] = null;
+    delete this.store[key];
+  }
+
+  public deepClone(obj : any) {
+    let tmpStore = obj;
+
+    if (!tmpStore || typeof tmpStore !== 'object') { return tmpStore; }
+
+    for (let key in tmpStore) {
```
Comment (AndyOGo) [inline src/store/index.ts]: May switch to `Object.keys`

File: src/store/index.ts (hunk 1 above)
Comment (LucaMele) [reply src/store/index.ts]:
```
  public addToStore(key : string, value: any) {
    if (!key || !value) {
      return false;
    }
    if (this.store[key]) {
      if (typeof value === 'object') {
        this.store[key] = this.deepClone({
          ...this.store[key],
          ...value,
        });
      } else {
        this.store[key] = value;
      }
    } else {
      if (typeof value === 'object') {
        this.store[key] = this.deepClone(Array.isArray(value) ? [...value] : {...value});
      } else {
        this.store[key] = this.deepClone(value);
      }
    }
  }
```

Comment (LucaMele) [reply src/store/index.ts]: needed a bit more logic :)

File: src/store/index.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+export default class Store implements IImmutable {
+
+  private store = {};
+
+  public addToStore(key : string, value: any) {
+    if (!key || !value) {
+      return false;
+    }
+    if (this.store[key]) {
+      if (typeof value === 'object') {
```
Comment (AndyOGo) [inline src/store/index.ts]: I would cache this check and reuse it by `const isObject = typeof value === 'object'`

Comment (LucaMele) [reply src/store/index.ts]: nice

**Pull request merged**